torch/csrc/autograd/python_function.cpp

```diff
@@ -133,7 +134,13 @@ auto PyFunction::apply(variable_list&& inputs) -> variable_list {
 
   THPObjectPtr apply_fn(PyObject_GetAttrString(obj, "apply"));
   if (!apply_fn) throw python_error();
-  THPObjectPtr r(PyObject_CallObject(apply_fn, pyInputs.get()));
+
+  PyObject* callRes;
+  {
+    SwapPyThreadState swapThreadState;
+    callRes = PyObject_CallObject(apply_fn, pyInputs.get());
+  }
```

Comment:
Nice :) Though I wonder why you couldn't have assigned `callRes` to the `THPObjectPtr` to start with; assignment is defined on THPObjectPtr.